### PR TITLE
UX updates for Prefab Override prototype

### PR DESCRIPTION
**Pull request by chiyenteng**
Signed-off-by: chiyenteng <82238204+chiyenteng@users.noreply.github.com>

Updates UI/UX for Prefab Override prototype.

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/Editor/EditorPreferenceGetter.h
Comment (srikappa-amzn) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Editor/EditorPreferenceGetter.h:11]: A similar file was recently added to handle prefab editor preferences. You can merge these changes into that file to keep all editor prefab preferences in one place : https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabEditorPreferences.h

File: Code/Editor/ViewportTitleDlg.ui
Comment (srikappa-amzn) [inline Code/Editor/ViewportTitleDlg.ui:59]: are these ux changes okay to go in with or without overrides?

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutliner.qss
Comment (srikappa-amzn) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutliner.qss:33]: I cannot find these being used anywhere in this PR or in existing code. Can be deleted?

File: AutomatedTesting/Registry/editorpreferences.setreg
Comment (srikappa-amzn) [inline AutomatedTesting/Registry/editorpreferences.setreg:5]: This is the setreg that control viewport visual feedback mechanism (blurring everything and highlighting focused instance). We shouldn't be turning this on and leave it to the viewport team to enable this when they are ready. Please remove this.

**Review by srikappa-amzn — CHANGES_REQUESTED** [review]

File: AutomatedTesting/Registry/editorpreferences.setreg
Comment (cgalvan) [reply AutomatedTesting/Registry/editorpreferences.setreg]: +1

File: Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/BreadCrumbs.cpp
Comment (cgalvan) [inline Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/BreadCrumbs.cpp:308]: Is this meant for just the prefab overrides UX or for all breadcrumbs?

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp:2260]: This check should still be as close to the top as possible to early-out, so right inside the `if (event->type() == QEvent::MouseButtonPress)` line

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp:908]: Is this going to break the functionality of clicking on a selected item to rename?

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.cpp
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.cpp:450]: Can we document these hard-coded numbers?

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.cpp
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.cpp:58]: This can be removed, it's not being used

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.h
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.h:78]: This can be removed, it's not being used

**Review by cgalvan — CHANGES_REQUESTED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.cpp
Comment (chiyenteng) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabUiHandler.cpp:447]: Sorry I really don't know how @AMZN-daimini decided these numbers.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutliner.qss
Comment (srikappa-amzn) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutliner.qss:33]: If they are not used anywhere in the code, I don't see any point in keeping them around. We shouldn't be merging in unused code when we aren't sure what it does.

**Review by srikappa-amzn — APPROVED** [review]
Looks good to me. 

Please do sanity ux testing before merging to dev to make sure existing ux is unaffected with these changes. Also, please remove unused code in EntityOutliner.qss

**Review by cgalvan — APPROVED** [review]

**Pull request merged**
